spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/opentelemetry/OpenTelemetryAutoConfiguration.java

```diff
@@ -80,6 +80,9 @@ private Resource toResource(Environment environment, OpenTelemetryProperties pro
 			.asMap();
 		attributes.computeIfAbsent("service.name", (key) -> getApplicationName(environment));
 		attributes.computeIfAbsent("service.group", (key) -> getApplicationGroup(environment));
+		// See https://github.com/spring-projects/spring-boot/issues/44411 for potential
+		// information about deprecation of "service.group" attribute
```

Comment:
I think that instead of reusing the existing `getApplicationGroup` method, a new method,
`getServiceNamespace`, can be introduced. This would make it possible to mark the
existing method with the `@Deprecated` annotation and include appropriate javadoc.

```java
private String getServiceNamespace(Environment environment) {
    return environment.getProperty("spring.application.group");
}
/**
 * @deprecated ...
 */
@Deprecated(since = "3.5.0", forRemoval = true)
private String getApplicationGroup(Environment environment) {
}
```

Using the `@Deprecated` annotation would make it easier to ensure that the
`service.group` property is deleted in future versions.


The same could be done in `OtlpMetricsPropertiesConfigAdapter`

Reply:
Thanks @nosan, i think that's a good idea.